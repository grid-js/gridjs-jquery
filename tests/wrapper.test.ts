/// <reference path="../src/jquery.d.ts"/>

import { Grid } from "gridjs";

const flushPromises = () => new Promise(setImmediate);

describe("wrapper", () => {
  it("should call the render method", () => {
    document.body.innerHTML = '<div id="wrapper"></div>';

    const grid = jest.spyOn(Grid.prototype, "render");

    $("div#wrapper").Grid({
      data: [[1, 2, 3]],
    });

    expect(grid).toBeCalled();
  });

  it("should render the table", async () => {
    document.body.innerHTML = '<div id="wrapper"></div>';

    $("div#wrapper").Grid({
      data: [[1, 2, 3]],
    });

    await flushPromises();

    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it("should render a table with custom headers", async () => {
    document.body.innerHTML = '<div id="wrapper"></div>';

    $("div#wrapper").Grid({
      data: [[1, 2, 3]],
      columns: [
        "a",
        "b",
        {
          name: "c",
          width: "200px",
        },
      ],
    });

    await flushPromises();

    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it("should convert the existing table", async () => {
    document.body.innerHTML =
      '<div id="wrapper">' +
      '<table id="myTable">' +
      "<thead><tr><th>Name</th><th>Age</th></tr></thead>" +
      "<tbody>" +
      "<tr><td>john</td><td>25</td></tr>" +
      "<tr><td>mark</td><td>55</td></tr>" +
      "<tr><td>oj</td><td>420</td></tr>" +
      "</tbody>" +
      "</table>" +
      "</div>";

    $("table#myTable").Grid();

    await flushPromises();

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
