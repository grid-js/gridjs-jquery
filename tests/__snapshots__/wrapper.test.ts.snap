// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapper should convert the existing table 1`] = `"<div id=\\"wrapper\\"><table id=\\"myTable\\" style=\\"display: none;\\"><thead><tr><th>Name</th><th>Age</th></tr></thead><tbody><tr><td>john</td><td>25</td></tr><tr><td>mark</td><td>55</td></tr><tr><td>oj</td><td>420</td></tr></tbody></table><div><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-wrapper\\" style=\\"height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"height: auto;\\"><thead class=\\"gridjs-thead\\"><tr class=\\"gridjs-tr\\"><th data-column-id=\\"name\\" class=\\"gridjs-th\\"><div class=\\"gridjs-th-content\\">Name</div></th><th data-column-id=\\"age\\" class=\\"gridjs-th\\"><div class=\\"gridjs-th-content\\">Age</div></th></tr></thead><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td data-column-id=\\"name\\" class=\\"gridjs-td\\">john</td><td data-column-id=\\"age\\" class=\\"gridjs-td\\">25</td></tr><tr class=\\"gridjs-tr\\"><td data-column-id=\\"name\\" class=\\"gridjs-td\\">mark</td><td data-column-id=\\"age\\" class=\\"gridjs-td\\">55</td></tr><tr class=\\"gridjs-tr\\"><td data-column-id=\\"name\\" class=\\"gridjs-td\\">oj</td><td data-column-id=\\"age\\" class=\\"gridjs-td\\">420</td></tr></tbody></table></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div></div></div>"`;

exports[`wrapper should render a table with custom headers 1`] = `"<div id=\\"wrapper\\"><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-wrapper\\" style=\\"height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"height: auto;\\"><thead class=\\"gridjs-thead\\"><tr class=\\"gridjs-tr\\"><th data-column-id=\\"a\\" class=\\"gridjs-th\\"><div class=\\"gridjs-th-content\\">a</div></th><th data-column-id=\\"b\\" class=\\"gridjs-th\\"><div class=\\"gridjs-th-content\\">b</div></th><th data-column-id=\\"c\\" class=\\"gridjs-th\\" style=\\"width: 200px;\\"><div class=\\"gridjs-th-content\\">c</div></th></tr></thead><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">1</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">2</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">3</td></tr></tbody></table></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div></div>"`;

exports[`wrapper should render the table 1`] = `"<div id=\\"wrapper\\"><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-wrapper\\" style=\\"height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"height: auto;\\"><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td class=\\"gridjs-td\\">1</td><td class=\\"gridjs-td\\">2</td><td class=\\"gridjs-td\\">3</td></tr></tbody></table></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div></div>"`;
